File: tools/sap/src/main/java/top/bettercode/summer/tools/sap/connection/SapSysException.java
package top.bettercode.summer.tools.sap.connection;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

/**
 * @author Peter Wu
 */
@ResponseStatus(HttpStatus.BAD_REQUEST)
public class SapSysException extends RuntimeException {

  private static final long serialVersionUID = 1L;

  public SapSysException(String message) {
    super("SAP系统：" + message);
  }

}

File: tools/sap/src/main/kotlin/top/bettercode/summer/tools/sap/connection/SapSysException.kt
package top.bettercode.summer.tools.sap.connection

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus

/**
 * @author Peter Wu
 */
@ResponseStatus(HttpStatus.BAD_REQUEST)
class SapSysException(message: String?) : RuntimeException("SAP系统：$message") {
    companion object
}
